pkg/app/server.go

```diff
@@ -323,10 +336,104 @@ func RunKubeStateMetrics(ctx context.Context, opts *options.Options) error {
 	if err := g.Run(); err != nil {
 		return fmt.Errorf("run server group error: %v", err)
 	}
+
 	klog.InfoS("Exited")
 	return nil
 }
 
+func pollForCacheUpdates(
```

Comment:
could you move that inside of another file? The server.go has a tendency to grow quickly so it is better to limit what we put inside it.

Maybe moving it inside of the discovery package would be better suited.